runtime/smithy-client/common/src/aws/smithy/kotlin/runtime/client/config/ClientSettings.kt

```diff
@@ -0,0 +1,20 @@
+package aws.smithy.kotlin.runtime.client.config
+
+import aws.smithy.kotlin.runtime.InternalApi
+import aws.smithy.kotlin.runtime.config.EnvironmentSetting
+import aws.smithy.kotlin.runtime.config.enumEnvSetting
+import aws.smithy.kotlin.runtime.config.intEnvSetting
+
+@InternalApi
+public object ClientSettings {
+    /**
+     * The maximum number of request attempts to perform. This is one more than the number of retries, so
+     * maxAttempts = 1 will have 0 retries.
+     */
+    public val MaxAttempts: EnvironmentSetting<Int> = intEnvSetting("SDK_MAX_ATTEMPTS", "sdk.maxAttempts")
+
+    /**
+     * Which RetryMode to use for the default RetryPolicy, when one is not specified at the client level.
+     */
+    public val RetryMode: EnvironmentSetting<RetryMode> = enumEnvSetting<RetryMode>("SDK_RETRY_MODE", "sdk.retryMode")
```

Comment:
question: should we use the `KOTLIN_SDK` namespace (i.e `KOTLIN_SDK_RETRY_MODE`) to avoid conflicts?

Reply:
No, I think that would make common behavior among multiple generic SDKs difficult. For instance, if Rust or JavaScript eventually supported this setting in a non-AWS SDK then each language would have different setting names and clients wouldn't be able to set a single value to set the setting across SDK languages.

Reply:
Maybe `aws.smithy` namespace and the `AWS_SMITHY_` environment variable prefix should be claimed/used.